src/OrleansSQLUtils/Storage/Provider/IStorageHashPicker.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Orleans;
+using Orleans.Runtime;
+using System.Collections.Generic;
+
+
+namespace OrleansSQLUtils.Storage
+{
+    /// <summary>
+    /// A picker to choose from provided hash functions.
+    /// </summary>
+    /// <remarks>Provider agility to update or change hashing functionality for both built-in and custom operations.</remarks>
+    public interface IStorageHasherPicker
```

Comment:
My thinking was: Doing like this allows doesn't make tie Orleans core hash function strongly to this storage provider and ultimately to the stored data. If Orleans ever modifies its hash, something like this would be needed anyhow. Other concern is that one might prefer clearly more collision resistant hash in case the input isn't controlled by Orleans (denial of service, for instance).